napalm_panos/panos.py

```diff
@@ -534,3 +534,107 @@ def get_interfaces(self):
+
+            <response status="success">
+              <result>
+                <ifnet>
+                  <entry>
+                    <name>ethernet1/5</name>
+                    <zone/>
+                    <fwd>N/A</fwd>
+                    <vsys>1</vsys>
+                    <dyn-addr/>
+                    <addr6>
+                      <member>fe80::d61d:71ff:fed8:fe14/64</member>
+                      <member>2001::1234/120</member>
+                    </addr6>
+                    <tag>0</tag>
+                    <ip>169.254.0.1/30</ip>
+                    <id>20</id>
+                    <addr>
+                      <member>1.1.1.1/28</member>
+                    </addr>
+                  </entry>
+                  {...}
+                </ifnet>
+                <hw>
+                  {...}
+                </hw>
+              </result>
+            </response>
+            '''
+            intf = parsed_intf_dict['name']
+            _ip_info = {intf: {}}
+
+            v4_ip = parsed_intf_dict.get('ip')
+            secondary_v4_ip = parsed_intf_dict.get('addr')
+            v6_ip = parsed_intf_dict.get('addr6')
+
+            if v4_ip != 'N/A':
+                address, pref = v4_ip.split('/')
+                _ip_info[intf].setdefault('ipv4', {})[address] = {'prefix_length': int(pref)}
+
+            if secondary_v4_ip is not None:
+                members = secondary_v4_ip['member']
+                if not isinstance(members, list):
+                    # If only 1 secondary IP is present, xmltodict converts field to a string, else
+                    # it converts it to a list of strings.
+                    members = [members]
+                for address in members:
+                    address, pref = address.split('/')
+                    _ip_info[intf].setdefault('ipv4', {})[address] = {'prefix_length': int(pref)}
+
+            if v6_ip is not None:
+                members = v6_ip['member']
+                if not isinstance(members, list):
+                    # Same "1 vs many -> string vs list of strings" comment.
+                    members = [members]
+                for address in members:
+                    address, pref = address.split('/')
+                    _ip_info[intf].setdefault('ipv6', {})[address] = {'prefix_length': int(pref)}
+
+            # Reset dictionary if no addresses were found.
+            if _ip_info == {intf: {}}:
+                _ip_info = {}
+
+            return _ip_info
+
+        ip_interfaces = {}
+        cmd = "<show><interface>all</interface></show>"
+
+        try:
+            self.device.op(cmd=cmd)
+            interface_info_xml = xmltodict.parse(self.device.xml_root())
+            interface_info_json = json.dumps(
+                interface_info_xml['response']['result']['ifnet']['entry']
+            )
+            interface_info = json.loads(interface_info_json)
+        except KeyError:
+            return ip_interfaces
```

Comment:
not following this one? When would `<show><interface>all</interface></show>` return a condition that meets this?

Reply:
good point, I was following the original implementation which had a similar approach, but missed giving that some thought.

Seems the better approach would be remove the `try/except` altogether, and let the call fail if something goes wrong. Come to think of it that should IMO be the same approach for the new `get_interfaces()` too.